library/ssl_tls13_server.c

```diff
@@ -1883,8 +1882,12 @@ static int ssl_tls13_prepare_new_session_ticket( mbedtls_ssl_context *ssl,
                 (mbedtls_ssl_ciphersuite_t *) ssl->handshake->ciphersuite_info;
     psa_hash_alg = mbedtls_psa_translate_md( ciphersuite_info->mac );
     hash_length = PSA_HASH_LENGTH( psa_hash_alg );
-    if( hash_length == -1 )
+    if( hash_length == -1 ||
+        hash_length > (int)sizeof( session->resumption_key ) )
```

Comment:
```suggestion
        (size_t)hash_length > sizeof( session->resumption_key ) )
```
safer I'd say

Reply:
I agree - also, if for some reason (e.g. weird bug in called code) `hash_length` comes back as `< 0` but `!= -1`, the case to `size_t` will make it very big, so this condition will be true, and we will error out